lib/msal-browser/src/request/RequestHelpers.ts

```diff
@@ -81,6 +85,19 @@ export async function initializeBaseRequest(
         validatedRequest.requestedClaimsHash = await hashString(request.claims);
     }
 
+    const hasRequestInstanceAware =
+        validatedRequest.extraQueryParameters &&
+        validatedRequest.extraQueryParameters.hasOwnProperty("instance_aware");
+
+    // Set instance_aware flag if config auth param is set
+    if (!hasRequestInstanceAware && config.auth.instanceAware) {
+        validatedRequest.extraQueryParameters =
+            validatedRequest.extraQueryParameters || {};
+        validatedRequest.extraQueryParameters["instance_aware"] = String(
```

Comment:
Rather than changing extraQueryParameters I would suggest doing this at url construction time (requestParameterBuilder)

Reply:
+1. Since we are introducing adding first class support for this, RequestParameterBuilder is the right place.

Reply:
Moved it to RequestParameterBuilder.